components/org.wso2.identity.webhook.wso2.event.handler/src/main/java/org/wso2/identity/webhook/wso2/event/handler/api/builder/WSO2TokenEventPayloadBuilder.java

```diff
@@ -241,4 +256,57 @@ private Application buildApplicationFromResourceId(EventData eventData, String r
         }
         return null;
     }
+
+    private String resolveUserResidentTenantDomain(String userResidentOrganizationId) {
+
+        if (StringUtils.isBlank(userResidentOrganizationId)) {
+            return null;
+        }
+
+        try {
+            return WSO2EventHookHandlerDataHolder.getInstance().getOrganizationManager()
+                    .resolveTenantDomain(userResidentOrganizationId);
+        } catch (OrganizationManagementException e) {
+            LOG.debug("Error while resolving tenant domain for user resident organization: " +
+                    userResidentOrganizationId, e);
+        }
+
+        return null;
+    }
+
+    private Organization buildUserResidentOrganizationForOrgUser(String userResidentOrganizationId,
+                                                                 String userResidentTenantDomain) {
+
+        if (StringUtils.isBlank(userResidentOrganizationId) || StringUtils.isBlank(userResidentTenantDomain)) {
+            return null;
+        }
+
+        try {
+            MinimalOrganization userResidentOrganization =
+                    WSO2EventHookHandlerDataHolder.getInstance().getOrganizationManager()
+                            .getMinimalOrganization(userResidentOrganizationId, userResidentTenantDomain);
+            if (userResidentOrganization != null) {
+                return new Organization(userResidentOrganization.getId(),
+                        userResidentOrganization.getName(),
+                        userResidentOrganization.getOrganizationHandle(),
+                        userResidentOrganization.getDepth());
+            }
+        } catch (OrganizationManagementException e) {
+            LOG.debug("Error while resolving retrieving user resident organization: " +
+                    userResidentOrganizationId, e);
+        }
+        return null;
```

Comment:
Let's add a debug log that no organization found for the given user resident organization id

Reply:
Fixed 7802dae7753b368615c4d3d8683cfbb12f3f91e4